program/rust/src/rust_oracle.rs

```diff
@@ -23,5 +33,94 @@ pub fn update_version(
     instruction_data: &[u8],
 ) -> OracleResult {
     panic!("Need to merge fix to pythd in order to implement this");
-    Ok(0) //SUCCESS
+    Ok(SUCCESS)
 }
+
+
+/// initialize the first mapping account in a new linked-list of mapping accounts
+/// accounts[0] funding account           [signer writable]
+/// accounts[1] new mapping account       [signer writable]
+pub fn init_mapping(
```

Comment:
please check that this is a line-by-line translation of the C

Reply:
i fiddled with this a bit per the comments, so please check again. it's a little harder to tell that it's a line-by-line translation now